services/core-web/src/HOC/ApplicationGuard.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from "react";
+import Loading from "@/components/common/Loading";
+import hoistNonReactStatics from "hoist-non-react-statics";
+import {
+  fetchImportedNoticeOfWorkApplication,
+  fetchOriginalNoticeOfWorkApplication,
+  fetchImportNoticeOfWorkSubmissionDocumentsJob,
+} from "@mds/common/redux/actionCreators/noticeOfWorkActionCreator";
+import * as routes from "@/constants/routes";
+import { clearNoticeOfWorkApplication } from "@mds/common/redux/actions/noticeOfWorkActions";
+import NOWStatusIndicator from "@/components/noticeOfWork/NOWStatusIndicator";
+import NullScreen from "@/components/common/NullScreen";
+import { fetchDraftPermitByNOW, fetchPermits } from "@mds/common/redux/actionCreators/permitActionCreator";
+import { useHistory, useLocation, useParams } from "react-router-dom";
+import { useAppDispatch } from "@mds/common/redux/rootState";
+/**
+ * @constant ApplicationGuard - Higher Order Component that fetches applications, and handles all common logic between the application routes.
+ * If users try to access an "Administrative Amendment" via the Notice of work Flow, they will be redirected.
+ * If users try to access an "Notice of Work" via the Administrative Amendment Flow, they will be redirected.
+ * If the Guid in the url changes, HOC looks for that change and updates.
+ * If the guid is not an application_guid, HOC shows nullScreen
+ * HOC handles all fixedTop/Scroll logic and passes flag to children
+ *
+ */
+
+const ApplicationGuard = (WrappedComponent) => {
+  const applicationGuard = (props) => {
+    const dispatch = useAppDispatch();
+    const history = useHistory();
+    const location = useLocation<{ applicationPageFromRoute?: any }>();
+    const { id, tab } = useParams<{ id: string, tab: string }>();
+    const showNullScreen = !location.state && !id;
+    const [fixedTop, setFixedTop] = useState(false);
+    const fixedTopRef = React.useRef(fixedTop);
```

Comment:
I ended up converting this one to TS first. Aaaand I should probably have brought the fixedTopRef bit over to ScrollSidePageWrapper. Kept it because it was done and it works.